File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {

        HashMap<Character, Character> maps1 = new HashMap<>();
        HashMap<Character, Character> maps2 = new HashMap<>();

        for(int i = 0; i < s.length(); i++) {

            char ch1 = s.charAt(i);
            char ch2 = t.charAt(i);

            if(maps1.containsKey(ch1) && maps1.get(ch1) != ch2) {
                return false;
            }

            if(maps2.containsKey(ch2) && maps2.get(ch2) != ch1) {
                return false;
            }

            maps1.put(ch1, ch2);
            maps2.put(ch2, ch1);
        }

        return true;
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        int n = s.size();
    int m = t.size();
    unordered_map<char, vector<int>> m1;
    unordered_map<char, vector<int>> m2;

    for (int i = 0; i < n; i++) {
        m1[s[i]].push_back(i);
    }

    for (int i = 0; i < m; i++) {
        m2[t[i]].push_back(i);
    }

    for (int i = 0; i < n; i++) {
        if (m1[s[i]] != m2[t[i]]) {
            return false;
        }
    }

    return true;

    }
};